js/index.js

```diff
@@ -28,6 +36,45 @@ $(function() {
       }
     };
 
+    var setupSession = function() {
+        function guid() {
+          function s4() {
+            return Math.floor((1 + Math.random()) * 0x10000)
+              .toString(16)
+              .substring(1);
+          }
+          return s4() + s4() + '-' + s4() + '-' + s4() + '-' +
+            s4() + '-' + s4() + s4() + s4();
+        }
+
+        var sessionId = $.QueryString['session'];
+        if (!sessionId) {
```

Comment:
If the string is empty, this will take the `if` branch, which is fine, but `addQueryStringField` then results in `&session=&session=...` (minor point)